src/posts/lucky_0.25.cr

```diff
@@ -0,0 +1,370 @@
+
+    ### Model Association updates
+
+    When you're using a RDBMS like postgres, table associations become very important for properly structuring your data. There were several bugs fixed when it comes to
+    associations giving us a lot more power and control over our data.
+
+    To start, we will look at the `belongs_to` association method. It's pretty common to name your method the same as the model it's referencing, but in some cases,
+    you may want your association method to be named slightly different like this:
+
+    ```crystal
+    class Employee < BaseModel
+      table do
+        belongs_to boss : Manager
+      end
+    end
+    ```
+
+    But this would throw an error due to how the query methods were generated. We would be expecting to generate a method named `where_managers`, but that doesn't exist here.
+
+    With this release, we no longer make the assumption on what you're naming your associations. This also means that a small convention must be followed where your `where_` query
+    methods will append the name of the association.
+
+    ```crystal
+    EmployeeQuery.new.where_boss(ManagerQuery.new)
+    ```
+
+    To go along with the `belongs_to`, we've also fixed some bugs related to using `has_many through`. This update will require a slightly different syntax.
+
+    ```crystal
+    # Before update
+    class User < BaseModel
+      table do
+        has_many posts : Post
+
+        # you specified the Symbol of the has_many method to query through
+        has_many comments : Comment, through: :posts
+      end
+    end
+
+    # After update
+    class User < BaseModel
+      table do
+        has_many posts : Post
+
+        # you specify an Array(Symbol) where the first item is the method to query through
+        # and the second item is that association's method.
+        has_many comments : Comment, through: [:posts, :comments]
+      end
+    end
+    ```
+
+    This means you can also now do a has_many though a has_many through 😬
+
+    ### Models now support SQL VIEW
+
+    Generally when we think of models, we think of database tables. The models even have a `table()` method to denote this!
+    Well now there's a `view()` method as well!
+
+    > SQL VIEWs are like tables, but generally READ-ONLY, and have no primary key. Their data is usually aggregated from other table sources.
+
+    ```crystal
+    class AdminUser < BaseModel
+      view do
+        column username : String
+        column promoted_to_admin_on : Time
+      end
+    end
+    ```
+
+    All of the columns for your `view` must be manually defined. These models do not come with any sort of `primary_key`, or `id` methods. This also means that methods like
+    `AdminUserQuery.find()`, and `admin_user.delete` won't work since these all rely on there being an `id` method. You must implement any of these methods yourself that you may
+    need.
```

Comment:
```suggestion
    All of the columns for your `view` must be manually defined. These models do not come with any sort of `primary_key` or `timestamps` by default. If your view has these columns, you will need to explicitly add them.
    View models will not have a `SaveOperation` defined as they are meant to be read-only, and if no primary key is added, some features will be missing from the model and `BaseQuery`.
    For example, `AdminUserQuery.find()` and `admin_user.reload` won't work since these rely on an `id` method.
    You must implement any of these methods yourself if you need them.
```

Trying to make it clear that some features could be missing because they don't have a primary_key but others just aren't available (SaveOperation)